catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java

```diff
@@ -163,8 +179,21 @@ public String toString() {
    *
    * @return query request
    */
-  private QueryRequest createQueryRequestForAllWorkspaceMetacards() {
-    return createQueryRequest(filterService.buildWorkspaceTagFilter());
+  private QueryRequest createQueryRequestForSubscribedToWorkspaces() {
+    Set<String> ids = new HashSet<>();
+    try {
+      List<Map<String, Object>> subscriptions =
+          persistentStore.get(SubscriptionsPersistentStore.SUBSCRIPTIONS_TYPE);
+      ids = subscriptions.stream().map(s -> s.get(ID_FIELD).toString()).collect(Collectors.toSet());
+    } catch (PersistenceException e) {
+      LOGGER.warn("Failed to get subscriptions for workspaces: {}", e.getMessage());
+      LOGGER.debug("Failed to get subscriptions for workspaces: {}", e.getMessage(), e);
+    }
+    if (ids.size() == 0) {
+      return null;
+    }
+
+    return createQueryRequest(createFilters(ids));
```

Comment:
✏️ Looks like the [`createFilters` method ](https://github.com/codice/ddf/blob/master/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java#L196-L204) Creates a `(id1 AND workspacetag) OR (id2 and workspacetag) ...`, we could just or all the id's and AND it with  a single workspacetag (instead of having an `AND workspacetag` filter with every single id) if we're worried about ever coming close to the 1024 boolean clause limit in solr. 

Beyond whats necessary for this though so this don't need to do now if its not a non issue.

Reply:
yeah, good point, we can clean up that logic